src/main/java/com/officelife/actions/TakeItem.java

```diff
@@ -21,25 +21,31 @@ public TakeItem(State state, Class<T> itemClass) {
 
   @Override
   public boolean accept() {
-    // TODO there may be many items on this location?
 
     Optional<Coords> currentCoords = state.world.actorLocation(state.person.id());
 
     if (!currentCoords.isPresent()) {
       throw new RuntimeException("person " + state.person.id() + " is nowhere");
     }
 
-    Item item = state.world.items.get(state.world.itemLocations.get(currentCoords.get()));
-    if (!itemClass.isInstance(item)) {
-      throw new RuntimeException("item is not of class " + itemClass);
-    }
+    String itemId = state.world.itemsAtLocation(currentCoords.get())
+            .stream()
+            .filter(id -> {
+              Item itemAtLocation = state.world.items.get(id);
+
+              return itemClass.isInstance(itemAtLocation);
+            })
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Unable to find item of " + itemClass));
```

Comment:
Hmm, this may not necessarily be a runtime error, because someone may have taken the item in the time since this agent committed to getting it.